packages/frontend/src/hooks/useChartColorConfig.tsx

```diff
@@ -1,10 +1,53 @@
 import { FeatureFlags, Series } from '@lightdash/common';
 import { useFeatureFlagEnabled } from 'posthog-js/react';
-import { useCallback, useState } from 'react';
+import { createContext, FC, useCallback, useContext, useState } from 'react';
 import { EChartSeries } from './echarts/useEchartsCartesianConfig';
 
 export type SeriesLike = EChartSeries | Series;
 
+interface ChartColorMappingContextProps {
+    colorMappings: Map<string, string>;
+}
+
+const ChartColorMappingContext =
+    createContext<ChartColorMappingContextProps | null>(null);
+
+/**
+ * Exposes a map of identifier->color values, which can be shared across
+ * a context, for shared color assignment.
+ */
+export const ChartColorMappingContextProvider: FC<
+    React.PropsWithChildren<{}>
+> = ({ children }) => {
+    const [colorMappings] = useState(new Map<string, string>());
```

Comment:
Using a mutable map (or any mutable data structure) directly as state in React goes against the core principles of React's state management.

Whenever you call `colorMappings.set` (from `useChartColorConfig`), React's re-rendering will bail on that because the reference remains the same. So, it acts like a `useRef`.

You can either change it to something like this:

```jsx
const [colorMappings, setColorMappings] = useState(new Map<string, string>());

const addColor = (identifier, color) => {
  setColorMappings(new Map(colorMappings.set(identifier, color)));
}
```

and expose `addColor` from the provider...

...or just use an object, which might make more sense here unless you really want to use a Map:

```jsx
const [colorMappings, setColorMappings] = useState<Record<string, string>>({});

const addColor = (identifier, color) => {
  setColorMappings({ ...colorMappings, [identifier]: color });
}
```

Reply:
thanks for explaining 💟 and now it makes sense, although it's not clear when reading the code. I think either using `useRef` will make it explicit, or adding a comment to explain the reasoning behind it.